erpnext/manufacturing/doctype/work_order/work_order.py

```diff
@@ -252,7 +258,13 @@ def on_cancel(self):
 		self.validate_cancel()
 
 		frappe.db.set(self,'status', 'Cancelled')
-		self.update_work_order_qty_in_so()
+		prod_plan = frappe.get_doc('Production Plan', self.production_plan)
+		pp_ref = prod_plan.prod_plan_ref
+		if pp_ref:
+			self.update_work_order_qty_in_combined_so(cancel = True)
+		else:
+			self.update_work_order_qty_in_so()
+			
```

Comment:
Same code has added in the on_submit method better create separate method for this portion and call it from on_submit as well on_cancel.